src/Raven.Server/Documents/Sharding/ShardedDatabaseContext.Streaming.cs

```diff
@@ -158,34 +160,47 @@ await foreach (var result in results.WithCancellation(combinedState.Cancellation
                     pagingContinuation);
             }
 
-            public async IAsyncEnumerable<Document> GetDocumentsAsync(CombinedReadContinuationState documents, ShardedPagingContinuation pagingContinuation, string resultPropertyName = "Results")
+            public class DocumentLastModifiedComparer : Comparer<ShardStreamItem<BlittableJsonReaderObject>>
             {
-                await foreach (var result in PagedShardedDocumentsByLastModified(documents, resultPropertyName, pagingContinuation))
+                public override int Compare(ShardStreamItem<BlittableJsonReaderObject> x, ShardStreamItem<BlittableJsonReaderObject> y)
                 {
-                    yield return result.Item;
+                    if (x?.Item == null)
+                        return 1;
+                    if (y?.Item == null)
+                        return 1;
+                    try
+                    {
+                        return x.Item.GetMetadata().GetLastModified() < y.Item.GetMetadata().GetLastModified() ? 1 : -1;
```

Comment:
return `y.Item.GetMetadata().GetLastModified() - x.Item.GetMetadata().GetLastModified()` would be simpler